### PR TITLE
Reduce volume of 'Waiting for current token' logs, which were introduced in v1.109.0.

### PR DESCRIPTION
**Pull request by reivilibre**
Introduced in: #17215

This caused us a minor bit of grief as the volume of logs produced was much higher than normal

### CODE REVIEW

File: synapse/types/__init__.py [hunk 1]
```diff
@@ -1082,6 +1096,15 @@ def is_before_or_eq(self, other_token: "StreamToken") -> bool:
 
         return True
 
+    def __str__(self) -> str:
+        return (
+            f"StreamToken(room: {self.room_key}, presence: {self.presence_key}, "
+            f"typing: {self.typing_key}, receipt: {self.receipt_key}, "
+            f"account_data: {self.account_data_key}, push_rules: {self.push_rules_key}, "
+            f"to_device: {self.to_device_key}, device_list: {self.device_list_key}, "
+            f"groups: {self.groups_key}, un_partial_stated_rooms: {self.un_partial_stated_rooms_key})"
```
Comment (erikjohnston) [inline synapse/types/__init__.py:1105]: Why is this a better representation than the default? Isn't this just replace e.g. `presence=5` with `presence: 5`?

**Review by erikjohnston — COMMENTED** [review]
I'm somewhat cautious about changing the default `__str__` functions, since its so easy to forget about updating them

File: synapse/types/__init__.py (hunk 1 above)
Comment (reivilibre) [reply synapse/types/__init__.py:1105]: 2 things to cut down on the noise:
- removes `_token` suffix on all the fields
- better formatting for the dict, which by default is `immutabledict({..})`

**Review by erikjohnston — APPROVED** [review]
I'm mildly against having non-default `__str__` implementations, but won't block this. The alternative I guess is to change the `StreamToken` to change the field names or something

**Pull request merged**